gravityflow/gravityflow
# Set default `number` to 2000 when get users in the Assignee field/setting

This PR fixes an issue that after PR #219, sites with a large number of users cannot load (or load very slowly) step settings or assignee fields.

We now set the default number to `2000` and add a notice to settings description when we don't display a full list of users. We also properly handle a situation that prevents assignees from being removed after the user list gets modified.

## Screenshots
<img width="845" alt="Screenshot 2019-05-12 11 44 45" src="https://user-images.githubusercontent.com/12166/57577515-5f620600-74ab-11e9-9c59-e0ab8087d821.png">

<img width="1047" alt="Screenshot 2019-05-12 11 45 41" src="https://user-images.githubusercontent.com/12166/57577522-7e609800-74ab-11e9-8b84-9e95ceca678d.png">

## Testing instructions
1. You can use FakerPress to generate more than 2,000 users. Or just use `gravityflow_get_users_args` to set a default `number`, so you can see the new description in your assignee and routing assignees fields.
2. Set some steps with assignees, pick some users from the bottom of the user list. Now use the `gravityflow_get_users_args` filter to change the `number` to `5` (to make sure the assignees previously picked won't display in the user list). Now reload the step settings, we expect to see them still show up in the user list.
3. Be sure you test both assignees and routing assignees. Also test the Notification step and different notification tabs in Approval/User Input steps etc..

## Note
Gravity Flow PDF extension loads feed settings as the step settings so we need to manually add the new description to it. I will create another PR for it.
